src/pcnt.rs

```diff
@@ -0,0 +1,578 @@
+
+    /// Enable PCNT input filter
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn filter_enable(&self) -> Result<(), EspError> {
+        unsafe { esp!(pcnt_filter_enable(self.unit)) }
+    }
+
+    /// Disable PCNT input filter
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn filter_disable(&self) -> Result<(), EspError> {
+        unsafe { esp!(pcnt_filter_disable(self.unit)) }
+    }
+
+    /// Set PCNT filter value
+    /// 
+    /// @param filter_val PCNT signal filter value, counter in APB_CLK cycles.
+    ///       Any pulses lasting shorter than this will be ignored when the filter is enabled.
+    ///       @note
+    ///       filter_val is a 10-bit value, so the maximum filter_val should be limited to 1023.
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn set_filter_value(&self, value: u16) -> Result<(), EspError> {
+        unsafe { esp!(pcnt_set_filter_value(self.unit, value)) }
+    }
+
+    /// Get PCNT filter value
+    /// 
+    /// returns
+    /// - i16
+    /// - EspError
+    pub fn get_filter_value(&self) -> Result<u16, EspError> {
+        let mut value = 0u16;
+        unsafe {
+            esp!(pcnt_get_filter_value(
+                self.unit,
+                &mut value as *mut u16
+            ))?;
+        }
+        Ok(value)
+    }
+
+    /// Set PCNT counter mode
+    /// 
+    /// @param channel PCNT channel number
+    /// @param pos_mode Counter mode when detecting positive edge
+    /// @param neg_mode Counter mode when detecting negative edge
+    /// @param hctrl_mode Counter mode when control signal is high level
+    /// @param lctrl_mode Counter mode when control signal is low level
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn set_mode(&self,
+        channel: PcntChannel,
+        pos_mode: PcntCountMode,
+        neg_mode: PcntCountMode,
+        hctrl_mode: PcntControlMode,
+        lctrl_mode: PcntControlMode) -> Result<(), EspError> {
+        unsafe {
+            esp!(pcnt_set_mode(self.unit, channel.into(), pos_mode.into(), neg_mode.into(), hctrl_mode.into(), lctrl_mode.into()))
+        }
+    }
+
+    /// Add ISR handler for specified unit.
+    /// 
+    /// This ISR handler will be called from an ISR. So there is a stack
+    /// size limit (configurable as \"ISR stack size\" in menuconfig). This
+    /// limit is smaller compared to a global PCNT interrupt handler due
+    /// to the additional level of indirection.
+    /// 
+    /// # Safety
+    ///
+    /// Care should be taken not to call STD, libc or FreeRTOS APIs (except for a few allowed ones)
+    /// in the callback passed to this function, as it is executed in an ISR context.
+    /// 
+    /// @param callback Interrupt handler function.
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    #[cfg(feature = "alloc")]
+    pub unsafe fn subscribe(&self, callback: impl FnMut(u32) + Send + 'static) -> Result<(), EspError> {
```

Comment:
A very small nit: can you - rather than using `impl FnMut` switch to `callback: C` and then `C: FnMut(u32)...`.